client/src/utils/uploadImage.js

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+import { Button } from "../components/Button";
+import styled from "styled-components/macro";
+import { ImageInput } from "../components/Input";
+import { ImagePreview } from "../components/ImagePreview";
+
+export default function UploadImage() {
+  const [imageInput, setImageInput] = useState("");
+  const [previewSource, setPreviewSource] = useState("");
+
+  const handleImageChange = (event) => {
+    const file = event.target.files[0];
+    previewFile(file);
+    setImageInput(event.target.value);
+  };
+
+  const previewFile = (file) => {
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.onloadend = () => {
+      setPreviewSource(reader.result);
+    };
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!previewSource) return;
+    uploadImage(previewSource);
+  };
+
+  const uploadImage = async (base64EncodedImage) => {
+    try {
+      await fetch("/api/upload", {
+        method: "POST",
+        body: JSON.stringify({ data: base64EncodedImage }),
+        headers: { "Content-Type": "application/json" },
+      });
+      setImageInput("");
+      setPreviewSource("");
+    } catch (error) {
+      console.error(error);
```

Comment:
Can be done in upcoming pull requests, but it would be great to have a visible notification about errors in the UI (and not only in the logs).

Reply:
i'll have to deal with this in this PR because i just realised, even though the image is uploaded to cloudinary my fetch fails